doc/developer/design/20231211_datadog_sink.md

```diff
@@ -0,0 +1,162 @@
+<details>
+<summary>Fields</summary>
+    
+| Field             | Type      | Definition|
+|-------------------|-----------|-----------|
+| metric [required] | String    | The name of the timeseries.                                                                                              |
+| points [required] | [Object]  | Points relating to a metric. All points must be objects with timestamp and a scalar value (cannot be a string). Timestamps should be in POSIX time in seconds, and cannot be more than ten minutes in the future or more than one hour in the past. |
+| resources         | [Object]  | A list of resources to associate with this metric.                                                                       |
+| source_type_name  | String    | The source type name.                                                                                                    |
+| tags              | [String]  | A list of tags associated with the metric.                                                                              |
+| type              | enum      | The type of metric. The available types are 0 (unspecified), 1 (count), 2 (rate), and 3 (gauge). Allowed enum values: 0, 1, 2, 3                                                        |
+| unit              | String    | The unit of point value.                                                                                                 |
+| interval          | Int64     | If the type of the metric is rate or count, define the corresponding interval.                                          |
+| metadata          | [Object]  | Metadata for the metric.                                                                                                 |
+    
+</details>
+
+All of the fields except the timestamp are set by the user. Materialize will set the timestamp before retrieving the metrics from its source.
+
+#### Timestamp
+
+In Materialize, we have two values to set the timestamp field: `mz_now()` and `now()`. I believe `now()` is enough. The integrations we recommend to our users today use the wall-clock time as a timestamp, so I don’t see any strong reason to switch to `mz_now()` unless we want correctness.
+
+While we could allow the user to set the timestamp, it would make the experience harder. Collections like `mz_cluster_replica_utilization` do not include a timestamp, forcing users to create one themselves. *I do not discard the chance to make it optional.*
+
+### Materialize source structure
+
+A Datadog sink needs a _source_: a table or view — and must have a structure that matches [Datadog's API submit metrics endpoint](https://docs.datadoghq.com/api/latest/metrics/#submit-metrics) fields.
+
+A table in Materialize acting as a source for the sink must have the following structure:
+
+```sql
+CREATE TABLE metrics (
+    -- Required:
+    metric TEXT,
+    resources JSONB,
+    value FLOAT,
+    -- Optional:
+    type INT,
```

Comment:
A little ugly to expose the raw int here. The ideal would be a PostgreSQL-style enum (https://www.postgresql.org/docs/current/datatype-enum.html), but of course we don't support those yet. In the meantime, a `text` field with the set of allowed values documented (`count`, `rate`, `gauge`) might be more user friendly.